src/accelerate/utils/environment.py

```diff
@@ -16,7 +16,7 @@
 import platform
 import subprocess
 import sys
-from distutils import spawn
+from shutil import which
```

Comment:
This won't quite do as Accelerate's minimum python version is 3.8 still. We need to do an if/else import check based on the python version and bring in the proper one :)